pythran/pythonic/types/ndarray.hpp

```diff
@@ -518,6 +519,15 @@ namespace pythonic {
                         return out;
                     }
 
+                /*
+                 * Default fast accessor when no fast overload are provided
+                 */
+                template <class Type>
+                    auto fast(Type && t) -> decltype(this->operator[](std::forward<Type>(t)))
+                    {
+                        return (*this)[std::forward<Type>(t)];
+                    }
+
```

Comment:
this seem dangerous to me. Your overload will take priority over conversion.

```c++
#include <cstdio>
#include <utility>

template<class T>
long foo(T&& t) { puts(__PRETTY_FUNCTION__); return std::forward<T>(t); }
long foo(long t) { puts(__PRETTY_FUNCTION__); return t;}
int main() {
  foo(1);
  foo(1L);
  return 0;
}
```
outputs
```
long foo(T &&) [T = int]
long foo(long
```